docs/articles/documentation/test-api/obtaining-data-from-the-client/examples-of-using-client-functions.md

```diff
@@ -98,74 +97,50 @@ test('My test', async t => {
-
-... and you need to obtain the text content of the first child node - *This is my tested page*. You can do this within a client function using the [childNodes](https://developer.mozilla.org/en-US/docs/Web/API/Node/childNodes) property. The following sample demonstrates this.
-
-```js
-import { Selector } from 'testcafe';
-import { ClientFunction } from 'testcafe';
-
-fixture `My Fixture`
-    .page `examplepage.html`;
-
-    const testedPage = Selector('#testedpage');
-
-    const getChildNodeText = ClientFunction(index => {
-        return getEl().childNodes[index].textContent;
-    },{ dependencies: { getEl: testedPage } });
-
-
-test('My Test', async t => {
-    await t.expect(getChildNodeText(0)).contains('This is my tested page.');
-});
-```
-
-> Note that the `getChildNodeText` client function uses the `testedPage` selector that is passed to it as a dependency. See [options.dependencies](README.md#optionsdependencies) for more information.
-
 ## Complex DOM Queries
 
 Client functions can be helpful for complex DOM queries. For example, a tested page contains a table with some data, and you need to validate data from two columns only. To do this, obtain data from these columns, push them to an array and then compare the returned array with the expected one.
 
 Below is a sample test for the [https://js.devexpress.com/](https://js.devexpress.com/) page that contains a grid. The sample demonstrates how to create a client function that returns an array containing data from the grid’s **Customer** and **Sale Amount** columns.
 
 ```js
-import { ClientFunction} from 'testcafe';
+import { ClientFunction } from 'testcafe';
 
 fixture `Get sale amount`
     .page('https://js.devexpress.com/');
 
-    const getSaleAmount = ClientFunction(() => {
-        const elements = document.querySelector('.dx-datagrid-rowsview').querySelectorAll('td:nth-child(3),td:nth-child(7)');
-        const array = [];
+    const getSalesAmount = ClientFunction(() => {
+        const grid      = document.querySelector('.dx-datagrid-rowsview');
+        const rowCount  = grid.querySelectorAll('.dx-data-row').length;
+        const sales     = grid.querySelectorAll('td:nth-child(3)');
+        const customers = grid.querySelectorAll('td:nth-child(7)');
 
-        for (let i = 0; i <= elements.length - 2; i+=2) {
-            const customerName  = elements[i+1].textContent;
-            const saleAmount = elements[i].textContent;
+        const array = [];
 
-            if (customerName && saleAmount)
-                array.push(`Customer ${customerName}: ${saleAmount}`);
+        for (let i = 0; i < rowCount; i++) {
+            array.push({
+                sales: sales[i].textContent,
+                customer: customers[i].textContent
+            });
         }
 
         return array;
     });
 
 test('My test', async t => {
-    console.log(await getSaleAmount());
+    const salesAmount = await getSalesAmount();
+
+    await t
+        .expect(salesAmount).eql([
```

Comment:
BTW best practice is to calculate CF value in assertion itself  `const salesAmount = await getSalesAmount();  await t.expect(salesAmount)` -> `await t.expect(getSalesAmount())`